src/app/main.py

```diff
@@ -23,12 +23,29 @@ def new_dataset(cloud_event):
     bucket_name = cloud_event.data["bucket"]
     filename = cloud_event.data["name"]
 
-    dataset = dataset_storage.get_dataset(filename=filename, bucket_name=bucket_name)
-
-    logger.info("Dataset obtained successfully.")
-    logger.debug(f"Dataset: {dataset}")
-
-    dataset_id = str(uuid.uuid4())
-    database.set_dataset(dataset_id=dataset_id, filename=filename, dataset=dataset)
-
-    logger.info("Dataset uploaded successfully.")
+    """
```

Comment:
Is this functionality already covered by a unit test? If not could one be added please

Reply:
@AND-JAshcroft This was covered by the test_set_dataset in database_test.py, the test that was missing was for the invalid JSON in dataset_storage.py. I've just added an invalid JSON input in conftest.py and added a test in dataset_storage_test.py, now the coverage is 100% for dataset_storage.py. Hope this looks alright now, thanks.